lib/ansible/module_utils/network/common/network.py

```diff
@@ -212,7 +212,7 @@ def get_resource_connection(module):
 
     capabilities = get_capabilities(module)
     network_api = capabilities.get('network_api')
-    if network_api == 'cliconf':
+    if network_api == 'cliconf' or network_api == 'nxapi':
```

Comment:
@mikewiebe Thanks for the fix.
Will you please update this to the following?
```
diff --git a/lib/ansible/module_utils/network/common/facts/facts.py b/lib/ansible/module_utils/network/common/facts/facts.py
index b2559ce91c..9fcc5319cb 100644
--- a/lib/ansible/module_utils/network/common/facts/facts.py
+++ b/lib/ansible/module_utils/network/common/facts/facts.py
@@ -85,6 +85,9 @@ class FactsBase(object):
         :param data: previously collected configuration
         :return:
         """
+        if self_connection not in ('network_cli', 'netconf'):
+            return
+
         if net_res_choices:
             if 'all' in net_res_choices:
                 net_res_choices.remove('all')
diff --git a/lib/ansible/module_utils/network/common/network.py b/lib/ansible/module_utils/network/common/network.py
index c8eb5308c4..d3abdb7819 100644
--- a/lib/ansible/module_utils/network/common/network.py
+++ b/lib/ansible/module_utils/network/common/network.py
@@ -212,7 +212,7 @@ def get_resource_connection(module):
 
     capabilities = get_capabilities(module)
     network_api = capabilities.get('network_api')
-    if network_api == 'cliconf':
+    if network_api in ('cliconf', 'nxapi', 'eapi'):
         module._connection = Connection(module._socket_path)
     elif network_api == 'netconf':
         module._connection = NetconfConnection(module._socket_path)
```

The first couple of lines of code in `get_network_resources_facts` method indicates that the resource module facts population in not invoked for the unsupported transports.

Reply:
@trishnaguha If you already have a PR in the works with these changes, I am happy to close mine and wait for yours.

Reply:
Sure I can open a PR.